src/problems/Albany_ThermalProblem.hpp

```diff
@@ -226,9 +276,11 @@ Albany::ThermalProblem::constructEvaluators(
     p->set<std::string>("Parameter Name", param_name);
     p->set<const Teuchos::ParameterList*>("Parameters List", &params->sublist("Parameters"));
     p->set<double>("Default Nominal Value", kappa[0]);
-    RCP<PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,Albany::ParamEnum,Albany::ParamEnum::Kappa_x>> ptr_kappa_x;
-    ptr_kappa_x = rcp(new PHAL::SharedParameter<EvalT,PHAL::AlbanyTraits,Albany::ParamEnum,Albany::ParamEnum::Kappa_x>(*p,dl));
-    fm0.template registerEvaluator<EvalT>(ptr_kappa_x);
+    if(!PHAL::is_param_available<EvalT>(fm0, param_name, dl->shared_param)) {
```

Comment:
Like in LandIce, all the plist construction/setup for all params can be moved inside their if blocks.

Note: this has nothing to do with perf, obviously. It just makes intent clearer, I think.